### PR TITLE
Add `EntityModel` class with Graph API CRU operations

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR adds the `EntityModel` class for creating, getting and updating `Entities`. 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202731581883433/f) _(internal)_

## 🔍 What does this change?

- Changes PersistedEntity in the rust API to not flatten the `identifier` as Utoipa doesn't support this
- Adds a new `EntityModel` class under `/api/src/models/knowledge/` directory

## ⚠️ Known issues

- We've added explicit NIL UUID references instead of relying on passed-in `accountId`s for the existing model classes. This needs to change and will allow us to refer to `created_by` workspace accounts in the future.

## 🐾 Next steps

- Implementing user entities [Asana task](https://app.asana.com/0/1200211978612931/1202693393444197/f) (_internal_)

## 🛡 What tests cover this?

- `packages/hash/integration/src/tests/model/knowledge/entity.model.test.ts`

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Recreate and migrate DB and run backend
     - From the `packages/graph/hash_graph/bin/hash_graph` path run:
     - `cargo make recreate-db && cargo make migrate-up`
     - `cargo run`
1.  In another terminal, from the `packages/hash/integration` path run `yarn jest "knowledge/*"`
1.  Confirm all tests pass

## 📹 Demo

image

### CODE REVIEW

File: packages/hash/api/src/model/ontology/entity-type.model.ts [hunk 1]
```diff
@@ -64,13 +65,15 @@ export default class {
...
     },
   ): Promise<EntityTypeModel> {
-    const { accountId, versionedUri } = params;
+    const { versionedUri } = params;
     const { data: schema } = await graphApi.getEntityType(versionedUri);
 
+    /** @todo: retrieve accountId from `graphApi.getEntityType` response */
+    const accountId = NIL_UUID;
+
     return new EntityTypeModel({ schema, accountId });
```
Comment (benwerner01) [inline packages/hash/api/src/model/ontology/entity-type.model.ts:77]: @Alfred-Mountfield @TimDiekmann do we have an existing ticket for this?

File: packages/graph/hash_graph/lib/graph/src/knowledge/entity/mod.rs [hunk 2]
```diff
@@ -81,7 +81,6 @@ impl PersistedEntityIdentifier {
 #[serde(rename_all = "camelCase")]
 pub struct PersistedEntity {
     inner: Entity,
-    #[serde(flatten)]
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/lib/graph/src/knowledge/entity/mod.rs:84]: Maybe we want to genuinely just flatten the fields into the struct then, not sure

File: packages/hash/api/src/model/ontology/entity-type.model.ts (hunk 1 above)
Comment (Alfred-Mountfield) [reply packages/hash/api/src/model/ontology/entity-type.model.ts:77]: https://app.asana.com/0/1201095311341924/1202770062090893/f

Comment (benwerner01) [reply packages/hash/api/src/model/ontology/entity-type.model.ts:77]: Danke :)

File: packages/hash/api/src/model/knowledge/entity.model.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+   */
+  static async create(
+    graphApi: GraphApi,
+    params: {
+      accountId: string;
+      properties: object;
+      entityType: EntityTypeModel;
+    },
+  ): Promise<EntityModel> {
+    const { accountId, entityType, properties } = params;
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/api/src/model/knowledge/entity.model.ts]: @benwerner01

**Review by Alfred-Mountfield — DISMISSED** [review]

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**